canonical-web-and-design/maas-ui
# fix(ui): handle undefined machine workload annotations

## Done

- Improve handling of undefined machine workload annotations
- Drive-by fix for instances tab link

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Set the MAAS_URL to karura (or a different 2.9 MAAS)
- Open the machine list and check that the app doesn't crash
- Check that there is no filter option for "Workloads"
- Go to a machine's details page and check that there is no "Workloads" card

## Fixes

Fixes #2114
